components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/WebhookManagementFacade.java

```diff
@@ -0,0 +1,350 @@
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.database.utils.jdbc.NamedJdbcTemplate;
+import org.wso2.carbon.database.utils.jdbc.exceptions.TransactionException;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.webhook.management.api.constant.ErrorMessage;
+import org.wso2.carbon.identity.webhook.management.api.exception.WebhookMgtException;
+import org.wso2.carbon.identity.webhook.management.api.model.Webhook;
+import org.wso2.carbon.identity.webhook.management.api.model.WebhookSearchResult;
+import org.wso2.carbon.identity.webhook.management.api.service.WebhookSubscriberService;
+import org.wso2.carbon.identity.webhook.management.internal.component.WebhookManagementComponentServiceHolder;
+import org.wso2.carbon.identity.webhook.management.internal.dao.WebhookManagementDAO;
+import org.wso2.carbon.identity.webhook.management.internal.util.WebhookManagementExceptionHandler;
+
+import java.util.List;
+
+/**
+ * Facade implementation for WebhookManagementDAO.
+ * This class coordinates between webhook subscriber services and webhook management DAO.
+ * It ensures that webhooks are first subscribed with external systems before being persisted.
+ */
+public class WebhookManagementFacade implements WebhookManagementDAO {
+
+    private static final Log LOG = LogFactory.getLog(WebhookManagementFacade.class);
+
+    private final WebhookManagementDAO webhookManagementDAO;
+
+    /**
+     * Constructor with WebhookManagementDAO.
+     *
+     * @param webhookManagementDAO WebhookManagementDAO instance.
+     */
+    public WebhookManagementFacade(WebhookManagementDAO webhookManagementDAO) {
+
+        this.webhookManagementDAO = webhookManagementDAO;
+    }
+
+    /**
+     * Create a new webhook subscription in the database.
+     * First checks if a webhook with same endpoint exists, then subscribes with the webhook subscriber service,
+     * and finally persists it using the DAO.
+     *
+     * @param webhook  Webhook subscription to be created.
+     * @param tenantId Tenant ID.
+     * @throws WebhookMgtException If an error occurs while creating the webhook subscription.
+     */
+    @Override
+    public void createWebhook(Webhook webhook, int tenantId) throws WebhookMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
+        WebhookSubscriberService subscriberService =
+                WebhookManagementComponentServiceHolder.getInstance().getWebhookSubscriberService();
+
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                if (webhook.getStatus() == null || webhook.getStatus().equals("ACTIVE")) {
+                    boolean subscribed = subscriberService.subscribe(webhook, tenantDomain);
```

Comment:
TODO: Check metadata to get the current publisher.
If Http publisher no subscription.

Reply:
Address via https://github.com/wso2/product-is/issues/24084